pkg/deviceplugins/common.go

```diff
@@ -127,20 +129,17 @@ func waitForGRPCServer(socketPath string, timeout time.Duration) error {
 
 // dial establishes the gRPC communication with the registered device plugin.
 func gRPCConnect(socketPath string, timeout time.Duration) (*grpc.ClientConn, error) {
-	c, err := grpc.Dial(socketPath,
-		grpc.WithInsecure(),
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
any preference to initialize a new context rather than pass through the main ctx via the handler?

Reply:
logic has been updated to store the main context into pcideviceclaim handler, and this is in turn stored in pcideviceplugin, for use when the grpc connection is initialized.